src/types/common.ts

```diff
@@ -36,6 +37,11 @@ export type HttpRequest = {
 };
 
 export type HttpRequestOptions = {
+  retry?: {
+    maxRetries?: number;
+    errorCodes?: string[];
+    methods?: HttpMethods[];
+  };
```

Comment:
tried to match undici API: https://undici.nodejs.org/#/docs/api/RetryAgent?id=parameter-retryhandleroptions

it doesn't expose `statusCodes`/`retryAfter` because the retry mechanism relies on node-fetch throwing where it can't access status code or headers.